### ISSUE TITLE
string literals can be compared with arbitrary strings

### ISSUE THREAD

**Issue by d180cf** [Microsoft/TypeScript#8055]
Here is a simple example:

```ts
function f(s: "foo" | "bar") {
  if (s == "baz") {
    ...
  }
}
```

As you see, there is a typo in "baz", but tsc doesn't catch that. On the other hand, if you try to `s == 123`, tsc will say that a string cannot be compared with a number.

**Comment by DanielRosenwasser** [comment]
Thanks for filing but this is a duplicate of #6149.
